### PR TITLE
Added Task Segmentation

### PR DESCRIPTION
**Pull request by fitzlithepius**
- Added new task type: segmentation.
- Added Evaluation Metrics: Dice Score.

### CODE REVIEW

File: multimedeval/task_families.py [hunk 1]
```diff
@@ -341,6 +342,106 @@ def evaluate(self, predictions):
...
+                scorer_args = {"num_classes": self.num_classes, "per_class": True}
+
+            dice_scorer = GeneralizedDiceScore(**scorer_args)
+
+            predicted_answers = torch.tensor(predicted_answers, dtype=torch.long)
+            ground_truth = torch.tensor(ground_truth, dtype=torch.long)
+
+            dice = dice_scorer(predicted_answers, ground_truth).item()
+
+            metrics = {"dice": dice, "labels": set(labels)}
```
Comment (corentin-ryr) [inline multimedeval/task_families.py:439]: What are the labels? 
It seems it could somehow be included in the logs but it is not a metric. Did I miss something here?

Comment (fitzlithepius) [reply multimedeval/task_families.py:439]: It's not a metric, that's true. 
According to the design we discussed yesterday, for example if a dataset have three _labels_ to segment on, we split them into three individual rows in the _self.dataset_. Then the final dice don't actually give information on how good the model performs on each _label_.

This design is probably not enough. But we probably need to figure out a way to keep the _label_ individual metrics?

**Review by corentin-ryr — APPROVED** [review]
LTGM

File: multimedeval/task_families.py (hunk 1 above)
Comment (fitzlithepius) [reply multimedeval/task_families.py]: Okay,  I think that could be done here in the task families, I will try to do that in a newer PR.
